addons/microsoft_calendar/models/res_users.py

```diff
@@ -38,11 +38,13 @@ def _get_microsoft_calendar_token(self):
     def _is_microsoft_calendar_valid(self):
         return self.sudo().microsoft_calendar_token_validity and self.sudo().microsoft_calendar_token_validity >= (fields.Datetime.now() + timedelta(minutes=1))
 
-    def _refresh_microsoft_calendar_token(self, service):
+    def _refresh_microsoft_calendar_token(self, service=None):
```

Comment:
Could you please make this change? :smile:
```suggestion
    def _refresh_microsoft_calendar_token(self, service='calendar'):
```

Reply:
I can add this default parameter. But that will cause a problem that string object does not have attribute `_get_microsoft_client_secret`

Reply:
Ahh, Now I understand. 
The fix is there. Please review it again.